mule-extensions-api/src/main/java/org/mule/runtime/extension/api/loader/util/InfrastructureTypeUtils.java

```diff
@@ -41,24 +46,37 @@
 import com.google.common.collect.ImmutableMap;
 
 /**
- * Mapping for types considered of "Infrastructure", of the {@link Class} of the infrastructure type and the {@link String} name
- * of it.
+ * Utilities for types considered of "Infrastructure".
  *
- * @since 4.0
+ * @since 4.8
  */
-public final class InfrastructureTypeMapping {
-
-  public static final String TLS_NAMESPACE_URI = "http://www.mulesoft.org/schema/mule/tls";
-
-  private static final Map<Class<?>, InfrastructureType> MAPPING = ImmutableMap.<Class<?>, InfrastructureType>builder()
-      .put(TlsContextFactory.class, new InfrastructureType(TLS_PARAMETER_NAME, 8))
-      .put(SourceTransactionalAction.class, new InfrastructureType(TRANSACTIONAL_ACTION_PARAMETER_NAME, 6))
-      .put(OperationTransactionalAction.class, new InfrastructureType(TRANSACTIONAL_ACTION_PARAMETER_NAME, 7))
-      .put(org.mule.sdk.api.tx.OperationTransactionalAction.class, new InfrastructureType(TRANSACTIONAL_ACTION_PARAMETER_NAME, 7))
-      .put(TransactionType.class, new InfrastructureType(TRANSACTIONAL_TYPE_PARAMETER_NAME, 9))
-      .put(SchedulingStrategy.class, new InfrastructureType(SCHEDULING_STRATEGY_PARAMETER_NAME, 10))
-      .put(ErrorMapping.class, new InfrastructureType(ERROR_MAPPINGS_PARAMETER_NAME, 11))
-      .build();
+public class InfrastructureTypeUtils {
```

Comment:
can we somehow rethink this as an object structure? instead of a utilitity method for each thing we need, obtain them through an object representung the infra type.

Reply:
Done.